bin/main.rb

```diff
@@ -1,46 +1,105 @@
 #!/usr/bin/env ruby
-def players
-  puts "Hello! who's the player one?, you will play with X."
-  p_one = gets
-  puts "Cool, and who's going to play with O?"
-  p_two = gets
-  puts "Nice to meet you #{p_one.chomp} and #{p_two.chomp}. Lets begin!"
-  [p_one, p_two]
+class Player
+  attr_accessor :name, :char
+  def initialize(char)
+    @char = char
+  end
+
+  def start
+    puts "Hello, what's your player name?"
+    @name = gets
+    puts "#{@name.chomp}, you will play as #{@char} \n\n"
+    puts
+  end
+end
+
+first = Player.new('X')
+first.start
+second = Player.new('O')
+second.start
+
+class GameDisplay
+  attr_accessor :position, :pos, :repeated
+  def initialize
+    @pos = ['-', '-', '-', '-', '-', '-', '-', '-', '-']
+    @position = 0
+    @choosed = []
+  end
+
+  def maps(current_ply)
+    @name = current_ply[0]
+    @character = current_ply[1]
+    puts "Now #{@name.chomp}, it's your turn, choose wisely \n\n"
+    puts " 1 | 2 | 3             #{@pos[0]} | #{@pos[1]} | #{@pos[2]} "
+    puts '---+---+---           ---+---+---'
+    puts " 4 | 5 | 6             #{@pos[3]} | #{@pos[4]} | #{@pos[5]} "
+    puts '---+---+---           ---+---+---'
+    puts " 7 | 8 | 9             #{@pos[6]} | #{@pos[7]} | #{@pos[8]} \n\n"
+    @is_valid = false
+    while @is_valid == false
+      @position = gets
+      if @position.to_i.zero? || @choosed.any? { |u| u == @position } || @position.to_i.between?(1, 9) == false
+        puts "Wait! Please enter a valid number\n"
+        redo
+      else
+        @choosed << @position
+        @pos[position.to_i - 1] = @character
+      end
+      return @is_valid = true
```

Comment:
[Optional] Because this is the last line in your method you don't need to add the `return` keyword.